integration/service/inspect_test.go

```diff
@@ -129,7 +129,7 @@ func newSwarm(t *testing.T) *daemon.Swarm {
 	return d
 }
 
-func serviceContainerCount(client client.ServiceAPIClient, id string, count uint64) func(log poll.LogT) poll.Result {
+func serviceContainerCount(client client.ServiceAPIClient, id string, count uint64, checkState bool, desiredState swarm.TaskState) func(log poll.LogT) poll.Result {
```

Comment:
I think we can remove these new arguments. Checking for running state should be fine in all cases.

Reply:
I'll try it out and see why it doesn't get to the running state.

Reply:
I tried this branch locally with this diff: 

```
diff --git a/integration/service/create_test.go b/integration/service/create_test.go
index 423b929d48..2f80c340b9 100644
--- a/integration/service/create_test.go
+++ b/integration/service/create_test.go
@@ -43,7 +43,7 @@ func TestCreateWithLBSandbox(t *testing.T) {
 	require.NoError(t, err)
 
 	serviceID := serviceResp.ID
-	poll.WaitOn(t, serviceContainerCount(client, serviceID, instances, true, swarm.TaskStateRunning))
+	poll.WaitOn(t, serviceContainerCount(client, serviceID, instances))
 
 	network, err := client.NetworkInspect(context.Background(), overlayID, types.NetworkInspectOptions{})
 	require.NoError(t, err)
diff --git a/integration/service/inspect_test.go b/integration/service/inspect_test.go
index 3d19d78275..c5adfa4bf0 100644
--- a/integration/service/inspect_test.go
+++ b/integration/service/inspect_test.go
@@ -37,7 +37,7 @@ func TestInspect(t *testing.T) {
 	require.NoError(t, err)
 
 	id := resp.ID
-	poll.WaitOn(t, serviceContainerCount(client, id, instances, false, swarm.TaskStateNew))
+	poll.WaitOn(t, serviceContainerCount(client, id, instances))
 
 	service, _, err := client.ServiceInspectWithRaw(ctx, id, types.ServiceInspectOptions{})
 	require.NoError(t, err)
@@ -129,7 +129,7 @@ func newSwarm(t *testing.T) *daemon.Swarm {
 	return d
 }
 
-func serviceContainerCount(client client.ServiceAPIClient, id string, count uint64, checkState bool, desiredState swarm.TaskState) func(log poll.LogT) poll.Result {
+func serviceContainerCount(client client.ServiceAPIClient, id string, count uint64) func(log poll.LogT) poll.Result {
 	return func(log poll.LogT) poll.Result {
 		filter := filters.NewArgs()
 		filter.Add("service", id)
@@ -140,11 +140,9 @@ func serviceContainerCount(client client.ServiceAPIClient, id string, count uint
 		case err != nil:
 			return poll.Error(err)
 		case len(tasks) == int(count):
-			if checkState {
-				for _, task := range tasks {
-					if task.Status.State != desiredState {
-						return poll.Continue("waiting for tasks to enter %v", desiredState)
-					}
+			for _, task := range tasks {
+				if task.Status.State != swarm.TaskStateRunning {
+					return poll.Continue("waiting for tasks to enter state %v", swarm.TaskStateRunning)
 				}
 			}
 			return poll.Success()
```

and it worked for me